src/psyclone/psyir/transformations/hoist_trans.py

```diff
@@ -112,6 +113,10 @@ def apply(self, node, options=None):
         # Place the assignment node before the loop.
         loop.parent.children.insert(loop.position, node)
 
+        # Remove loop if no more children are left.
+        if len(loop.loop_body.children) == 0:
```

Comment:
`if not loop.loop_body.children:` is slightly nicer?

Reply:
Done